src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Protocol;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public sealed class TemplateLocator
+    {
+        private DirectoryInfo _dotnetSdkTemplatesLocation;
+
+        public TemplateLocator()
+        {
+            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
+            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
+            {
+                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
+            }
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        {
+            if (_dotnetSdkTemplatesLocation == null)
+            {
+                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+            }
+
+            return LocalFolderUtility
+                .GetPackagesV2(_dotnetSdkTemplatesLocation.FullName, new NullLogger())
+                .Select(l => new OptionalSdkTemplatePackageInfo(l)).ToArray();
+        }
+
+        public string DotnetSdkVersionUsedInVs()
+        {
+            return "5.1.100";
```

Comment:
Should this be centralized eventually?

Reply:
This is a place holder implementation. Template engine currently don't need this. But it is part of the spec we decided on (for VS integration later)